### PR TITLE
Add dotall flag to regex functions

### PR DESCRIPTION
**Pull request by BaldFabi**
##### SUMMARY
This PR adds an additional flag to the regex functions. `dotall`/`re.S` was missing

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
core

### CODE REVIEW

**Review by mkrizek — CHANGES_REQUESTED** [review]
This needs a changelog fragment https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment and mention the new flag in the documentation https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#searching-strings-with-regular-expressions.

**Comment by BaldFabi** [comment]
Added the missing changes

File: changelogs/fragments/76886-add-dotall-flag-to-all-regex-functions.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - core - Add dotall flag to all regex functions
```
Comment (samccann) [inline changelogs/fragments/76886-add-dotall-flag-to-all-regex-functions.yml]:
```suggestion
  - core - Add dotall flag to all regex functions (https://github.com/ansible/ansible/pull/76886).
```

**Review by samccann — COMMENTED** [review]
Adding link to pr into the changelog fragment.

**Review by samccann — APPROVED** [review]
Docs portion LGTM.

**Review by mattclay — CHANGES_REQUESTED** [review]
We've discussed this amongst the core team and we're inclined to document the existing functionality in Python which supports setting flags as part of the pattern: https://docs.python.org/3/library/re.html#index-16

This would allow using all flags available to Python, not just the ones we've chosen to implement. This should be documented in a doc fragment so it can be added to each filter plugin using a regular expression.

We're also inclined to deprecate the existing flags arguments for `ignorecase` and `multiline`. If you're interested in doing that as well, we'd accept it in a separate PR.

**Comment by Akasurde** [comment]
superseded by https://github.com/ansible/ansible/pull/81016

**Pull request closed without merging**